metalarchives_importer.user.js

```diff
@@ -0,0 +1,242 @@
+// @version        2015.05.23
+// @description    import d'une release du site metal-archives vers musicbrainz
+// @downloadURL	   https://raw.githubusercontent.com/MetalDavidian/Import_Metal_Archives_releases_into_MB/master/Import_Metal_Archives_releases_into_MB.user.js
+// @update 		  https://raw.githubusercontent.com/MetalDavidian/Import_Metal_Archives_releases_into_MB/master/Import_Metal_Archives_releases_into_MB.user.js
+// @include        http://www.metal-archives.com/albums/*/*/*
+// @require        https://ajax.googleapis.com/ajax/libs/jquery/1.3.2/jquery.js
+// @require        lib/mbimport.js
+// ==/UserScript==
+$(document).ready(function () {
+    var release = retrieveReleaseInfo();
+    insertLink(release);
+});
+
+function setreleasedate(release, datestring) {
+    var patt = new RegExp('^d{4}$');
+    if (patt.exec(datestring)) {
+        release.year = datestring;
+    } else if (datestring.indexOf(',') != - 1) {
+        var commaindex = datestring.indexOf(',');
+        var d = new Date(datestring.substring(0, commaindex - 2) + datestring.substring(commaindex));
+        release.year = d.getFullYear();
+        release.month = d.getMonth()+1;
+        release.day = d.getDate();
+        console.log(release.day);
+    } else{
+        var d = new Date("2 "+datestring);
+        release.year = d.getFullYear();
+        release.month = d.getMonth()+1;
+        console.log(release.day);
+    }
+    return release;
+}
+
+function getGenericalData(){
+    var rdata=new Array();
+    var keydata = $('dl.float_left dt, dl.float_right dt').map(function(){
+        var s=$.trim($(this).text());
+        return s.substring(0,s.length-1);
+    } ).get();
+    var valuedata= $('dl.float_left dd,dl.float_right dd').map(function(){
+        return $.trim($(this).text());
+    } ).get();
+    for(i=0;i<keydata.length;i++){
+        rdata[keydata[i]]=valuedata[i];
+    }
+    console.log(rdata);
+    return rdata;
+}
+function retrieveReleaseInfo() {
+    var release = new Object();
+
+    release.artist_credit =new Array();
+    release.artist_credit.push({artist_name: $('h2.band_name').text(),credited_name: $('h2.band_name').text()});
+    release.title = $('h1.album_name').text();
+
+    var rdata=getGenericalData();
+
+    release = setreleasedate(release, rdata["Release date"]);
+    //todo add case for multiple labels if such a case exist
+    release.labels = new Array();
+    release.labels.push( { 
+        name: (rdata["Label"]== "Independent"?"[no label]":rdata["Label"]), 
+        catno: (rdata["Catalog ID"] == "N/A" ? "" :rdata["Catalog ID"]) 
+    } );
+
+    release.country = ""; // Worldwide
+
+
+    release.type =ReleaseTypes[rdata["Type"]][0];
+    release.secondary_types = new Array(ReleaseTypes[rdata["Type"]][1]);
+    release.status = 'official';
+
+    rdata.hasOwnProperty("Version desc.") &&  rdata["Version desc."].indexOf("Digipak")!=-1?release.packaging ="Digipak":release.packaging ="";
+    release.language = '';
```

Comment:
Ditto here, just skip the definition of language, script and barcode.
That would make the code smaller.